semantic_conventions/trace/messaging.yaml

```
@@ -0,0 +1,116 @@
+          type: string
+          brief: 'The version of the transport protocol.'
+          examples: '0.9.1'
+        - id: url
+          type: string
+          brief: 'Connection string.'
+          examples: ['tibjmsnaming://localhost:7222', 'https://queue.amazonaws.com/80398EXAMPLE/MyQueue']
+        - id: message_id
+          type: string
+          brief: 'A value used by the messaging system as an identifier for the message, represented as a string.'
+          examples: '452a7c7c7c7048c2f887f61572b18fc2'
+        - id: conversation_id
+          type: string
+          brief: >
+            The [conversation ID](#conversations) identifying the conversation to which the message belongs,
+            represented as a string. Sometimes called "Correlation ID".
+          examples: 'MyConversationId'
+        - id: message_payload_size_bytes
+          type: number
+          brief: >
+            The (uncompressed) size of the message payload in bytes.
+            Also use this attribute if it is unknown whether the compressed or uncompressed payload size is reported.
+          examples: 2738
+        - id: message_payload_compressed_size_bytes
+          type: number
+          brief: 'The compressed size of the message payload in bytes.'
+          examples: 2048
+      constraints:
+        - any_of:
+          - 'net.peer.name'
+          - 'net.peer.ip'
+        - include: network
+
+    - id: messaging.producer
+      prefix: messaging
+      extends: messaging
+      span_kind: producer
+      brief: 'Semantic convention for producers of messages sent to a messaging systems.'
+
+    - id: messaging.producer.synchronous
+      prefix: messaging
+      extends: messaging
+      span_kind: client
+      brief: >
+        Semantic convention for clients of messaging systems that produce messages
+        and synchronously wait for responses.
+
+    - id: messaging.consumer
+      prefix: messaging
+      extends: messaging
+      span_kind: consumer
+      brief: 'Semantic convention for a consumer of messages received from a messaging system'
+      attributes:
+        - id: operation
+          type:
+            allow_custom_values: false
+            members:
+              - id: receive
+                value: "receive"
+              - id: process
+                value: "process"
+          brief: >
+            A string identifying the kind of message consumption as defined in the
+            [Operation names](#operation-names) section above.
+            If the operation is "send", this attribute MUST NOT be set, since the
+            operation can be inferred from the span kind in that case.
+
+    - id: messaging.consumer.synchronous
+      prefix: messaging
+      extends: messaging.consumer
+      span_kind: server
+      brief: >
+        Semantic convention for servers that consume messages received from messaging systems
+        and always send back replies directed to the producers of these messages.
```

```suggestion
        and always send back replies directed to the producers of these messages.

```

Missing newlines at the end of files cause follow-up diffs adding to them to mistakenly show the last line as edited.